framework/include/constraints/AutomaticMortarGeneration.h

```diff
@@ -209,6 +241,26 @@ class AutomaticMortarGeneration : public ConsoleStreamInterface
     return msm_elem_to_info;
   }
 
+  int dim() const { return mesh.mesh_dimension(); }
+
+  /**
+   * @return The set of nodes on which mortar constraint is not active
+   */
+  const std::unordered_set<const Node *> & getInactiveLMNodes() const
+  {
+    return inactive_local_lm_nodes;
```

Comment:
In this case we're getting the actual geometric nodes that should be inactive, later on in MortarConstraintBase we use the system and variable information to get the actual degrees of freedom. For example in frictional contact each node will have a normal and frictional DoF associated with it. The other routine getInactiveLMElems does a similar thing but for elemental variables, we're just computing the geometric entities so that later we can find all the DoFs that we need to set to zero.